src/cpp/core/r_util/RActiveSessionStorage.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * RActiveSessionStorage.cpp
+ *
+ * Copyright (C) 2021 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#include <core/r_util/RActiveSessionStorage.hpp>
+#include <core/r_util/RActiveSessions.hpp>
+#include <core/FileSerializer.hpp>
+#include <core/Log.hpp>
+#include <core/system/Xdg.hpp>
+
+namespace rstudio {
+namespace core {
+namespace r_util {
+
+    FileActiveSessionStorage::FileActiveSessionStorage(const FilePath& activeSessionsDir) :
+        activeSessionsDir_ (activeSessionsDir)
+    {
+        Error error = activeSessionsDir_.ensureDirectory();
+        if(error)
+            LOG_ERROR(error);
+    }
+
+    Error FileActiveSessionStorage::readProperty(const std::string& id, const std::string& name, std::string* pValue)
+    {
+        std::set<std::string> propertyName = {name};
+        std::map<std::string, std::string> propertyValue{};
+        *pValue = "";
+
+        Error error = readProperties(id, propertyName, &propertyValue);
```

Comment:
Nit (for less lines - is a debatable nit)
```suggestion
        std::map<std::string, std::string> propertyValue{};
        *pValue = "";

        Error error = readProperties(id, { name }, &propertyValue);
```